### ISSUE TITLE
BUG: Datadog Agent Rate Limiting Logs

### ISSUE THREAD

**Issue by coreycarvalho** [department-of-veterans-affairs/notification-api#2111]
## Description

We are seeing rate limiting error messages form our Datadog Agent logs. This ticket seeks to investigate and remediate the issue.

![Image](https://github.com/user-attachments/assets/b606484e-b252-45d2-bbad-28d9ff97c1c5)

![Image](https://github.com/user-attachments/assets/a9a4d7d7-3342-4053-878c-6b287361f8f1)

- [x] Ticket is understood, and QA has been contacted (if the ticket has a QA label).
- [x] This work is added to the sprint review slide deck (key win bullet point and demo slide)

## Steps to Reproduce

1. Navigate to Datadog logs: `<env>-notification-api-datadog-log-group`
2. Search for `UTC | TRACE | ERROR` to view the log items in question

#### Low Impact

This does not impact any external clients, but does impact our ability to view logs (assuming this causes us to lose log items). Additionally, this seems to be happening in prod.

#### Medium Priority

Despite impact being low, this should be prioritized. It's important to trust our ability to log.

## Expected Behavior

- [x] The log items pictured above would not be present (i.e. our logs would not be rate limited)

**Comment by coreycarvalho** [comment]
I set logs to debug in dev and did not see any real pattern causing this. Some brief searching online pointed to the fact that it's because of the DD url we use (gov URL). I think the next step here is to open a support ticket with Datadog to get them to help us debug.

**Comment by coreycarvalho** [comment]
I reached out to support and they think that the root cause of the issue is from a misconfiguration in the default settings. The image below shows that two telemetry env vars are set to true by default, but those features are not enabled in a DD government cloud environment. DD support recommended we set those to false, but not until they do further investigation today.

<img width="762" alt="Image" src="https://github.com/user-attachments/assets/0463ce91-51a0-4b70-883f-9571500d654c" />

https://docs.datadoghq.com/tracing/configure_data_security/?tab=environmentvariables&site=gov#telemetry-collection
